accelerator/core/proxy_apis.c

```diff
@@ -99,24 +98,30 @@ static status_t api_find_transactions(const iota_client_service_t* const service
     goto done;
   }
   lock_handle_lock(&cjson_lock);
-  if (service->serializer.vtable.find_transactions_deserialize_request(obj, req) != RC_OK) {
+  if (core->iota_service.serializer.vtable.find_transactions_deserialize_request(obj, req) != RC_OK) {
     lock_handle_unlock(&cjson_lock);
     ret = SC_CCLIENT_JSON_PARSE;
     ta_log_error("%s\n", "SC_CCLIENT_JSON_PARSE");
     goto done;
   }
   lock_handle_unlock(&cjson_lock);
-
-  lock_handle_lock(&cjson_lock);
-  if (iota_client_find_transactions(service, req, res) != RC_OK) {
+  do {
+#ifdef CHRONICLE_ENABLE
+    if (db_client_find_transactions(&core->cr_service, req, res) == SC_OK) {
```

Comment:
Do we need lock here?
And are we able to integrate `line 110` with `line 116`? They are really simmilar tho

Reply:
No, we do not need lock here.
line 110 and 116 handle the proxy API to different hosts.